kotlinx-coroutines-core/jvm/test/exceptions/CoroutineExceptionHandlerJvmTest.kt

```diff
@@ -8,28 +8,15 @@ import kotlin.test.*
 
 class CoroutineExceptionHandlerJvmTest : TestBase() {
 
-    private val exceptionHandler = Thread.getDefaultUncaughtExceptionHandler()
-    private lateinit var caughtException: Throwable
-
-    @Before
-    fun setUp() {
-        Thread.setDefaultUncaughtExceptionHandler({ _, e -> caughtException = e })
-    }
-
-    @After
-    fun tearDown() {
-        Thread.setDefaultUncaughtExceptionHandler(exceptionHandler)
-    }
-
     @Test
     fun testFailingHandler() = runBlocking {
         expect(1)
-        val job = GlobalScope.launch(CoroutineExceptionHandler { _, _ -> throw AssertionError() }) {
-            expect(2)
-            throw TestException()
+        val caughtException = catchingUncaughtException {
+            GlobalScope.launch(CoroutineExceptionHandler { _, _ -> throw AssertionError() }) {
```

Comment:
Better to use any other type of exception as a second-level test exception, to avoid false connotations with assertions used in test for its true purpose.

Reply:
`IllegalStateException` is the one preferred by you and the project in general, I inferred.